### PR TITLE
Configure expression tests on CI

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
While working on another PR I noticed that we do not run it on CI, so I propose to start doing it.

### CODE REVIEW

File: metrics/ignores/platform-all.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "expression-tests/legacy/interval/composite-default": "https://github.com/mapbox/mapbox-gl-native/issues/12747",
     "expression-tests/legacy/interval/tokens-zoom": "https://github.com/mapbox/mapbox-gl-native/issues/12747",
     "expression-tests/resolved-locale/basic": "Even the 'en' locale may not be present on some test systems.",
+    "expression-tests/within/non-supported": "Something is wrong with GeoJSON geometry in test",
```
Comment (SiarheiFedartsou) [inline metrics/ignores/platform-all.json:19]: It throws exception while loading test information (even before we actually run it): 
```
2023-05-05 21:47:43.550 mbgl-expression-test[20437:12390485] [ERROR] {}[General]: Cannot load test data from '/Users/siarheifedartsou/work/maplibre-gl-native/metrics/integration/expression-tests/within/non-supported/test.json': Polygon must be described by 4 or more coordinate points. Improper nesting can also lead to this error. Double check that the coordinates are properly nested and there are 4 or more coordinates.
```

File: expression-test/expression_test_parser.cpp [hunk 2]
```diff
@@ -402,37 +402,42 @@ Ignores parseExpressionIgnores() {
...
-        Log::Error(Event::General, std::string("Can't convert inputs value for '") + path.string() + "'");
+        // Parse inputs
+        if (data.document.HasMember("inputs") && !parseInputs(data.document["inputs"], data)) {
+            Log::Error(Event::General, std::string("Can't convert inputs value for '") + path.string() + "'");
+            return std::nullopt;
+        }
+
+        return {std::move(data)};
+    } catch (const std::exception& ex) {
+        Log::Error(Event::General, std::string("Cannot load test data from '" + path.string() + "': " + ex.what()));
```
Comment (SiarheiFedartsou) [inline expression-test/expression_test_parser.cpp:438]: If there is a problem during load of test data we just return nullopt.

File: expression-test/main.cpp [hunk 3]
```diff
@@ -47,7 +47,11 @@ int main(int argc, char** argv) try {
         }
 
         if (!testRun) {
-            stats.errored.emplace_back(id);
+            if (shouldIgnore) {
```
Comment (SiarheiFedartsou) [inline expression-test/main.cpp:50]: If test cannot be loaded for whatever reason we still can ignore it.

**Comment by SiarheiFedartsou** [comment]
I think tests for CI were already red even before this change...

**Review by louwers — APPROVED** [review]
Awesome work!

Yes, about the Linux tests see #1070 and #1071.
I will have a look at them this week.

But in this case it is failing because of xvfb. Need to look into that as well...

**Pull request merged**